Makefile

```diff
@@ -487,13 +487,13 @@ $(SYSROOT_LIB)/libwasi-emulated-signal.a: $(LIBWASI_EMULATED_SIGNAL_OBJS) $(LIBW
 %.a:
 	@mkdir -p "$(@D)"
 	# On Windows, the commandline for the ar invocation got too long, so it needs to be split up.
-	$(AR) crs $@ $(wordlist 1, 199, $^)
-	$(AR) crs $@ $(wordlist 200, 399, $^)
-	$(AR) crs $@ $(wordlist 400, 599, $^)
-	$(AR) crs $@ $(wordlist 600, 799, $^)
+	$(AR) crs $@ $(wordlist 1, 199, $(sort $^))
+	$(AR) crs $@ $(wordlist 200, 399, $(sort $^))
+	$(AR) crs $@ $(wordlist 400, 599, $(sort $^))
+	$(AR) crs $@ $(wordlist 600, 799, $(sort $^))
 	# This might eventually overflow again, but at least it'll do so in a loud way instead of
 	# silently dropping the tail.
-	$(AR) crs $@ $(wordlist 800, 100000, $^)
+	$(AR) crs $@ $(wordlist 800, 100000, $(sort $^))
```

Comment:
Isn't it enough to simply sort when calling AR here?  

I don't see why the sorting needs to happen up in `LIBC_TOP_HALF_MUSL_SOURCES`.   IIUC those all get built in parallel anyway so it won't dictate the order of the building of the object files?

Also we should probably just switch to using a response file here rather than using multiple calls to AR.. but thats a different issue.

Reply:
Sorting the inputs to AR makes total sense.   We can land that now.

I would like to know why the filelists need to be sorted though.

Reply:
> I would like to know why the filelists need to be sorted though.

Sorting the objects was my first approach, but it didn't work, for some reason (maybe I missed one of the `$^`).
So, I moved to sorting the filelists (I thought the object list was generated from them, I am not good at reading `Makefile`s), and that was not enough, too.
Eventually I have tried to sort both, and it worked, so I have opened the PR with this change.

Only after your prompt I have checked again the sort applied only to the object list.
So, I think that the filelist doesn't need to be sorted after all (it doesn't hurt either, though :slightly_smiling_face: ).